[16.0][MIG] contract_payment_auto: Migration to 16.0

9184Lucky opened the pull request.

lasley committed:
[ADD] contract_auto: Automatic Payment For Contracts

pedrobaeza committed:
[FIX] contract_payment_auto: Switch test to HttpCase

pedrobaeza committed:
[FIX] contract_payment_auto: Fix RST syntax

moylop260 committed:
[FIX] contract_payment_auto: Fix freze wkhtmltopdf freezed

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-transbot committed:
OCA Transbot updated translations from Transifex

fcayre committed:
[FIX] contract_payment_auto: transaction create must always get a token (#167)

When a contrat had no payment token but the corresponding partner had
one, the transaction was created without an acquirer, leading to an
integrity error in postgres.

This change makes sure the token used to test the ability to pay an
invoice is passed along to the transaction creation call.

Tests were also added to check the ability to use the contract token if
present, but the partner's in the opposite case.

This change fixes #165.

oca-transbot committed:
OCA Transbot updated translations from Transifex

oca-travis committed:
[UPD] Update contract_payment_auto.pot

marylla committed:
Translated using Weblate (German)

Currently translated at 100.0% (25 of 25 strings)

Translation: contract-10.0/contract-10.0-contract_payment_auto
Translate-URL: https://translation.odoo-community.org/projects/contract-10-0/contract-10-0-contract_payment_auto/de/

OCA-git-bot committed:
[ADD] icon.png

norlinhenrik committed:
[MIG] contract_payment_auto: Migration to 12.0

norlinhenrik committed:
[MIG][FIX] contract_payment_auto: Migration to 12.0

norlinhenrik committed:
[MIG][FIX2] contract_payment_auto: Migration to 12.0

norlinhenrik committed:
[MIG][FIX3] contract_payment_auto: Migration to 12.0

fcayre committed:
[FIX] Make payment work with several invoices and contracts to be paid

The `_recurring_create_invoice` contract method returns multiple
invoices potentially coming from different contracts, which may have
different values for their `is_auto_pay` attribute.

Without this patch, this case simply crashes (a unit test with more than
one contract to be invoiced is a must-have in this module!).

fcayre committed:
[IMP] Fix small docstring typo

fcayre committed:
Fix tests

Change the mock method to patch the Contract class instead of the contract object:
patching the object is less reliable and depends on the implementation too much.
This implementation changed and the tests were no more working. Mocking the class
method fixes the problem.

fcayre committed:
Reactivate all tests and fix them

The payment acquirer API requires an implementation of a _get_feature_support
method. This is done using a temporary mock where required.

fcayre committed:
Do not crash if generated invoice has no line

fcayre committed:
[FIX] Add invoice on created payment transactions

Also pass the invoice to the transaction reference computation mecanism
so that it reflects the invoice's number.

Commit:
[IMP] contract_payment_auto: black, isort, prettier

Commit:
[MIG] contract_payment_auto: Migration to 16.0

Commit:
[IMP] contract_payment_auto: Replace model mock by model injection

This allows a cleaner overriding of _send_payment_request and improve
code readability.

Commit:
[IMP] contract_payment_auto: Improve test readability

Commit:
[IMP] contract_payment_auto: Move setUp to setUpClass to improve efficiency

Commit:
[IMP] contract_payment_auto: Make test class depend on TestContractBase

Initial purpose was to pass tests on a database with
l10n module installed but it also simplifies the setUpClass method.

rousseldenis commented:
/ocabot migration contract_payment_auto

rousseldenis commented:
@9184Lucky Could you improve your commits messages in order to make module name appear in it ?

Like `[IMP] contract_payment_auto: (...)`

![image](https://github.com/user-attachments/assets/16c322f4-09ae-42b1-baff-c56632a6a382)


Moreover, if you think some can be squashed with migration one, it could be great.

9184Lucky commented:
@rousseldenis Done ! I squashed all commit directly related to migration and fixed the other's messages.

If anyone feels that some other commit should be merged feel free to asks but i think the ones I left were improvement more than migration...

rousseldenis commented:
@9184Lucky Not yet pushed :sweat_smile:

9184Lucky commented:
Oups ^^'

9184Lucky commented:
@rousseldenis I pushed the new version but forgot to ping you, sorry !

rousseldenis commented:
> @rousseldenis I pushed the new version but forgot to ping you, sorry !

No problem. I'll check this later today

Honeyxilia reviewed (approved):
LGTM (code review)

rousseldenis reviewed (approved):
LGTM

rousseldenis commented:
/ocabot merge nobump
